src/assisted_test_infra/test_infra/controllers/proxy_controller/proxy_controller.py

```diff
@@ -6,82 +6,71 @@
 
 import consts
 from assisted_test_infra.test_infra import utils
+from assisted_test_infra.test_infra.controllers.containerized_controller import ContainerizedController
 from service_client import log
 
 
-class ProxyController:
+class ProxyController(ContainerizedController):
     PROXY_USER = "assisted"
     PROXY_USER_PASS = "redhat"
+    IMAGE = "quay.io/sameersbn/squid"
 
     def __init__(
         self,
         name=None,
         port=consts.DEFAULT_PROXY_SERVER_PORT,
         denied_port=None,
         authenticated=False,
-        dir=None,
         host_ip=None,
         is_ipv6=False,
     ):
+        super().__init__(name, port, self.IMAGE)
 
         if not name:
             self.address = ""
         else:
-            self.name = name
-            self.port = port
+            self._name = name
+            self._port = port
             self.authenticated = authenticated
-            self.image = "quay.io/sameersbn/squid"
-            self.dir = dir
+            self.image = ""
             self._is_ipv6 = is_ipv6
             self._set_server_address(host_ip)
             self._create_conf_from_template(denied_port=denied_port)
             self._create_user_file_for_auth()
-            self._run_proxy_server()
+            self.run()
```

Comment:
kind of weird to do the ``run`` from the c'tor. can we move it from this scope?

Reply:
I didn't want to change its behavior , you think it's ok to do it inside this PR?

Reply:
I think so. it's just moving one line